sdks/python/apache_beam/runners/direct/direct_runner.py

```diff
@@ -338,6 +337,7 @@ def run_pipeline(self, pipeline):
     from apache_beam.runners.direct.transform_evaluator import \
       TransformEvaluatorRegistry
     from apache_beam.testing.test_stream import TestStream
+    from apache_beam.metrics.execution import MetricsEnvironment
```

Comment:
Undo this change? Metrics should not be changed to depend on anything in the runners package. (Either that or the lazy import should be made there.)

Reply:
Or perhaps don't import metrics.execution when importing the public (write) Metrics api.

Reply:
That is reasonable, and I agree. I've fixed this by having a lazy import in the metrics execution module.
Currently, we're working under the idea that state sampler is the global context provider, so metrics rely on it. I'd think that makes it into a module that provides a service to other modules that require/handle context management. Though I'd agree that a more loose coupling would be good to have.